python/packages/autogen-agentchat/src/autogen_agentchat/conditions/_terminations.py

```diff
@@ -428,3 +436,53 @@ def _to_config(self) -> SourceMatchTerminationConfig:
     @classmethod
     def _from_config(cls, config: SourceMatchTerminationConfig) -> Self:
         return cls(sources=config.sources)
+
+
+class TextMessageTerminationConfig(BaseModel):
+    """Configuration for the TextMessageTermination termination condition."""
+    source: str | None = None
+    """The source of the text message to terminate the conversation."""
+
+
+class TextMessageTermination(TerminationCondition, Component[TextMessageTerminationConfig]):
+    """Terminate the conversation if a :class:`~autogen_agentchat.messages.TextMessage` is received.
+
+    This termination condition checks for TextMessage instances in the message sequence. When a TextMessage is found,
+    it terminates the conversation if either:
+    - No source was specified (terminates on any TextMessage)
+    - The message source matches the specified source
+
+    Args:
+        source (str | None, optional): The source name to match against incoming messages. If None, matches any source.
+            Defaults to None.
+    """
+
+    component_config_schema = TextMessageTerminationConfig
+    component_provider_override = "autogen_agentchat.conditions.TextMessageTermination"
+
+    def __init__(self, source: str | None = None) -> None:
+        self._terminated = False
+        self._source = source
+
+    @property
+    def terminated(self) -> bool:
+        return self._terminated
+
+    async def __call__(self, messages: Sequence[AgentEvent | ChatMessage]) -> StopMessage | None:
+        if self._terminated:
+            raise TerminatedException("Termination condition has already been reached")
+        for message in messages:
+            if isinstance(message, TextMessage) and (self._source is None or message.source == self._source):
+                self._terminated = True
+                return StopMessage(content="Stop message received", source="TextMessageTermination")
```

Comment:
The content should be updated to something like "TextMessage from .. was detected."